src/main/drivers/serial.c

```diff
@@ -88,7 +89,24 @@ void serialSetMode(serialPort_t *instance, portMode_e mode)
 
 void serialWriteBufShim(void *instance, const uint8_t *data, int count)
 {
-    serialWriteBuf((serialPort_t *)instance, data, count);
+    int remainingBytes = count;
+    int packetSize = 0;
+    while (remainingBytes > 0) {
+        if (remainingBytes > 20) {
+            packetSize = 20;
+            serialWriteBuf((serialPort_t *)instance, data + (count - remainingBytes), packetSize);
+        } else {
+            packetSize = remainingBytes;
+            serialWriteBuf((serialPort_t *)instance, data + (count - remainingBytes), packetSize);
+        }
+
+        // sleep 10 ms, it's safe for the BLE transfer speed limit
+        timeMs_t timeout = millis() + 10;
+        while (millis() < timeout) {
+        }
```

Comment:
I think that the `delay` method seen in other places do the same and is easier. 
https://github.com/betaflight/betaflight/blob/a24daa78d0fecce4c74c43789ea6c7b99c4d9ccf/src/main/drivers/compass/compass_qmc5883l.c#L130

Reply:
yes! get it!